### DISCUSSION TITLE
Incorrect Action scope on Resources with non-numerical primary key

### DISCUSSION THREAD

**Discussion by jcadsmurai** [laravel/nova-issues#5515] [Q&A] [answered]
- Laravel Version: 9.52.7
- Nova Version: 4.23.0
- PHP Version: 8.0.24
- Database Driver & Version: MySQL 8.0
- Operating System and Version: Running on Docker on Ubutnu / Running native on Debian
- Browser type and version: Google Chrome Version 112.0.5615.165

### Description:

When running an action to item(s) selected, the action is executed on all items (including all pages) instead of running only on the selected items. This issue occurs specifically with resources that have a **non-numerical primary key**.

### Detailed steps to reproduce the issue on a fresh Nova installation:

1. Create a table for a resource, defining the ID field as a string.
2. Populate the table with 213 rows of dummy data, ensuring that each row has an ID containing at least one alphabetic character.
3. Create the corresponding Eloquent model and Nova resource for the newly created table.
4. Define the ID field in the Nova resource using either the Text or ID field type (both have been confirmed to reproduce the problem).
5. Create an Action and assign it to the Resource.
6. Log in to Nova, navigate to the Resource, select one or more items.
7. Run the action.

### Observed Behavior:

The action is executed twice: first affecting 200 rows, and then affecting the remaining 13 rows.

### Expected Behavior:

The action should be executed only for the N selected items.

**Comment by crynobone** [comment]
Unable to reproduce the issue, please provide full reproducing repository based on fresh installation as suggested in the bug report template (or you can refer to https://github.com/nova-issues for example)

https://github.com/nova-issues/issue-5509/commit/049d26c4831c28f9dd5eb429fafb4b19f1b8a234

**Comment by jcadsmurai** [comment]
Thank you @crynobone 
I've been able to reproduce the error here https://github.com/jcadsmurai/nova-issue-5509/commit/a78d1e73df2d06fca687bfb8f37aa71fb2e61512

The main difference with your test is that the new resource does not use the `HasUuids` trait, and declares `public $incrementing = false;` on the Eloquent model.

**Comment by crynobone** [comment]
You should disable Action Logs if you only have few tables with such usage: https://nova.laravel.com/docs/4.0/actions/defining-actions.html#disabling-the-action-log

**Comment by jcadsmurai** [comment, marked as answer]
The problem was in the Eloquent model, the primary key had to be defined as a 'string' as stated in the Laravel documentation: https://laravel.com/docs/9.x/eloquent#primary-keys

Simply adding `protected $keyType = 'string';` fixed the problem, no other change on our codebase was needed.

Thanks for your time!
